From 1d2ae86e46909ea8883b7d2347340121703dd3c5 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:11:37 -0400
Subject: [ethernet] i40e/i40evf: Retrieve and store missing link config information

Message-id: <1409663591-32531-55-git-send-email-sassmann@redhat.com>
Patchwork-id: 90528
O-Subject: [RHEL7.1 PATCH 054/148] i40e/i40evf: Retrieve and store missing link config information
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Neerav Parikh <Neerav.Parikh@intel.com>

Some information returned via "Get Link Status" command was not
being cached in the struct i40e_link_status.
Add this so the driver can utilize this information as needed.

Change-ID: If084a0ae2a63b16b97572196b993742af2c67772
Signed-off-by: Neerav Parikh <neerav.parikh@intel.com>
Acked-by: Greg Rose <gregory.v.rose@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 6bb3f23c55bc8c14e9721c64bf51a8290ee50779)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40e_type.h b/drivers/net/ethernet/intel/i40evf/i40e_type.h
index 449a3be..51a6dee 100644
--- a/drivers/net/ethernet/intel/i40evf/i40e_type.h
+++ b/drivers/net/ethernet/intel/i40evf/i40e_type.h
@@ -173,6 +173,9 @@ struct i40e_link_status {
  u8 loopback;
  /* is Link Status Event notification to SW enabled */
  bool lse_enable;
+ u16 max_frame_size;
+ bool crc_enable;
+ u8 pacing;
 };
 
 struct i40e_phy_info {
-- 
1.7.1